central/summary/service/service_impl_test.go

```diff
@@ -15,14 +14,14 @@ func TestSummaryTypeToResourceMap(t *testing.T) {
 	summaryCountsRespType := reflect.TypeOf(v1.SummaryCountsResponse{})
 
 	for i := 0; i < summaryCountsRespType.NumField(); i++ {
-		name := summaryCountsRespType.Field(i).Name
+		field := summaryCountsRespType.Field(i)
 		// This ignores hidden metadata fields in the proto.
-		if strings.HasPrefix(name, "XXX_") {
+		if field.Tag.Get("protobuf") == "" {
```

Comment:
Do we need this at all with v2? This test might pass there

Reply:
We do because there are still some internal fields:
```
	state         protoimpl.MessageState
	sizeCache     protoimpl.SizeCache
	unknownFields protoimpl.UnknownFields
```
If we don't skip them, assertions will fail.

Reply:
I'd need deeper knowledge about the protobuf code generators here.
Looking at `summaryCountRespType`, it indeed looks like the fields of interest do indeed have `protobuf` annotations (or flags) with the current code generator. Will it be the case with the next protobuf code generator (I suppose the annotations come from the base go protobuf generator, but I could be wrong there)?

Reply:
Yes. Annotation is there. It is probably something that serializer requires.